CRM/MembershipExtras/Hook/Pre/MembershipEdit.php

```diff
@@ -157,6 +158,50 @@ public function preventExtendingPaymentPlanMembership() {
     }
   }
 
+  /**
+   * Prevents editing the active Membership if its contribution has another
+   * payment.
+   *
+   * Usually recording the payment for the membership's contribution means :
+   * - Activating the membership by changing its status from pending to active.
+   * - Renewing the membership by extending its end_date.
+   *
+   * At Compucorp, we use the "biz.jmaconsulting.lineitemedit" extension to add
+   * and edit the line items.
+   *
+   * Adding a line item changes the contribution status to "Partially Paid"
+   * and recording a payment for an active membership's contribution will renew
+   * it by default and extend its end_date, but this is undesirable behaviour
+   * and this method will prevent it.
+   */
+  public function preventEditingTheActiveMembershipIfItsContributionHasAnotherPayment() {
+    // Check if the membership is active and is not pay later.
+    $membershipCount = civicrm_api3('Membership', 'getcount', [
+      'sequential' => 1,
+      'active_only' => 1,
+      'id' => $this->id,
+      'is_pay_later' => 0,
```

Comment:
Civi  does not keep this field updated, so if you create a membership with :

1- Pending contribution (is_pay_later will be 1)
2- Complete the contribution (is_pay_later will remain 1)

so if someone then added new line item to the completed contribution then it won't be picked here, since is_pay_later is still = 1 since civi do not update it.

but to be honest, I don't it is needed, so If the membership is : 

1- active
2- and the contribution is currently partially paid and we are trying to record another payment.

then just skip updating its end date, since CiviCRM only extends the date if the payment is completed. But this might cause issues if we are using the line item editor with Payment plans, so I just asked Sheela on the ticket if we are using it with payment plans.

Reply:
Added a condition to skip payment plans.

Reply:
actually, I am rethinking this, and even the current approach might be an issue, here is why:

1- Create a Membership with a completed contribution (the membership status will be active).
2- Renew the membership manually but with a Pending contribution (the membership will remain active, and the end date will not get extended yet since the contribution is pending).
3- Partially pay the contribution (the membership is still active, and the end date still the same since Civi only extends the end date if the contribution got completed).
4- Pay the remaining amount of the contribution, now this step on CiviCRM core will complete the contribution, then extend the membership end date, but with this new logic, it will see an active membership with a partially paid contribution getting completed, and thus it will prevent the end date from getting extended, which is wrong.

at this point I am not sure how can we make this work, will need more thinking.